internal/db/definition_validation.go

```diff
@@ -978,3 +980,39 @@ func validateSchemaNameNotEmpty(
 
 	return nil
 }
+
+// validateCollectionMaterialized verifies that a non-view collection is materialized.
+//
+// Long term we wish to support this, however for now we block it off.
+func validateCollectionMaterialized(
+	ctx context.Context,
+	db *db,
+	newState *definitionState,
+	oldState *definitionState,
+) error {
+	for _, col := range newState.collections {
+		if len(col.QuerySources()) == 0 && !col.IsMaterialized {
+			return NewErrColNotMaterialized(col.Name.Value())
+		}
+	}
+
+	return nil
+}
+
+// validateCollectionMaterialized verifies that a materialized view has no ACP policy.
+//
+// Long term we wish to support this, however for now we block it off.
+func validateMaterializedHasNoACP(
```

Comment:
Suggestion: perhaps instead of `NoACP` in the name it can be `NoPolicy` or `NoACPPolicy` as no ACP can mean that acp is turned off completely.

Reply:
Not a problem.

- [ ] Rename `validateMaterializedHasNoPolicy`